### ISSUE TITLE
Cut down on Appveyor builds

### ISSUE THREAD

**Issue by pganssle** [dateutil/dateutil#529]
Currently this is the Appveyor build matrix:

```yaml
  matrix:
    - PYTHON: "C:/Python27"
    - PYTHON: "C:/Python27-x64"
    - PYTHON: "C:/Python33"
    - PYTHON: "C:/Python33-x64"
    - PYTHON: "C:/Python34"
    - PYTHON: "C:/Python34-x64"
    - PYTHON: "C:/Python35"
    - PYTHON: "C:/Python35-x64"
    - PYTHON: "C:/Python36"
    - PYTHON: "C:/Python36-x64"
```

I'm not sure I can recall a time when we had a failure on x64 that didn't also fail on x86, and Appveyor is often the slow step in the CI (I think they don't do concurrent builds). I'm thinking we should probably cut down and re-organize the build matrix, like this:

```yaml
  matrix:
    - PYTHON: "C:/Python27-x64"
    - PYTHON: "C:/Python36-x64"

    - PYTHON: "C:/Python27"
    - PYTHON: "C:/Python33"
    - PYTHON: "C:/Python34"
    - PYTHON: "C:/Python35"
    - PYTHON: "C:/Python36"
```

We'd be testing on all supported versions (except 3.2, which we've droped from these builds for whatever reason ages ago) on x86, and only test the latest two builds on x64, but the latest two x64 builds would be executed *first*.

The other option is to test x64 on all versions and only test the latest two with x86. The downside there is that I think 1. x86 is more likely to have problems than x64 and 2. you're *more likely* to be using x86 if you're also using an old version of Python.

I imagine when we add a binary component, we'll want to turn the full matrix back on again at least temporarily, since that's way more likely to have architecture + version corner cases.

**Comment by kirit93** [comment]
A simple fix for this could be to exclude tests we don't want. Eg. if we don't want Python33-x64 we can add 
```
matrix:
    exclude:
        - platform: x64
           PTYHON_VERSION: 33
```
And so on for the configurations we don't want to test.
Appveyor does concurrent builds based on the type of account you have. For the basic and professional it's just 1, but the premium account gives you 2 concurrent build and if there's an option for unlimited concurrent builds but that's around $300 per month.

**Comment by pganssle** [comment]
@kirit93 It's not really a question of implementation so much as whether these are actually *necessary*.

I'm not sure there's ever been a bug that fails on x86 that didn't also fail on x64 and vice-versa, so I think it's probably time to drop a bunch of them.

I think maybe we'll drop testing of `x86` builds except for the latest two versions.

**Comment by kirit93** [comment]
So Python2.7 and Python3.6 on x86 and everything on x64 will be tested right?
If using ```exclude``` is okay, I can give this a try.

**Comment by pganssle** [comment]
I haven't looked at the `appveyor.yml` file too deeply since @gaborbernat rearranged it to use `tox`. It was a much easier problem at the time since all platforms were explicitly listed.

I assume that `exclude` is fine. What would be the downside of that?

@kirit93 Feel free to make a PR with the changes. @gaborbernat If you have time do you mind reviewing Kirit's PR, since you know the new setup better than me?

**Comment by kirit93** [comment]
I'm not too sure what the downside of ```exclude``` would be if at all any. I'll submit a PR and let @gaborbernat review it and let me know if anything needs to change.

**Comment by pganssle** [comment]
Oops, this should have been closed by #579
